commitizen-tools/commitizen
# fix(git): force the default git locale on methods relying on parsing the output

## Description
This change ensure that methods relying on localized `git` output parsing use the default locale instead of the one defined by the user.

Currently, there was only on, `git.get_tags()` when no commit exists. This PR ensure the command is run with the default `C` locale (meaning no locale). Currently, it was failing to handle this case on non-english system and producing 2 test failures on non-english system.

## Checklist

- [x] Update the documentation for the changes (N/A)

## Expected behavior
Command output parsing works whatever the user locale.
Test runs and succeed whatever the user locale.

## Steps to Test This Pull Request
Define your environment locale (`LANG`/`LANGUAGE`/`LC_ALL` depending on the OS, shell ...) to anything else than `en_*` locale or `C` locale and then:
- run the test suite: it should succeed
- run the `cz changelog` command on a repository without commits: it should succeed


## Additional context
I choose to set the environment variables only on the command relying on localized output parsing. Benefit: any other call is localized and user can read potential failures in its own language. Drawback: devs should think about localization and forcing locale each time they want to parse a localized output.

Potential alternatives: 
- force all command to run in their default language
- force all `git` calls to run in `git` default locale
